### PR TITLE
add Key patch request to transfer to new key ring

### PR DESCRIPTION
**Pull request by gspurthi**
Associated with the issue: 473

Changes included in the PR: 
```
Added method TransferKeytoNewKeyRing
Updated unit tests for the method and the key structure change.
Fixed bug where the imported field is not shown in the response when false.
```

### CODE REVIEW

File: keys.go [hunk 1]
```diff
@@ -203,6 +207,26 @@ func (c *Client) createKey(ctx context.Context, key Key) (*Key, error) {
 	return &keysResponse.Keys[0], nil
 }
 
+// TransferKeyToNewKeyRing method transfers a key associated with one key ring to another key ring
+// For more information please refer to the link below:
+// https://cloud.ibm.com/docs/key-protect?topic=key-protect-grouping-keys#transfer-key-key-ring
+func (c *Client) TransferKeyToNewKeyRing(ctx context.Context, keyID, newKeyRingID string) (*Key, error) {
+	keyRingRequestBody := KeyRingID{
```
Comment (pbavari) [inline keys.go:214]: Validate keyID and newKeyRingID is valid string

File: keys.go [hunk 2]
```diff
@@ -203,6 +207,26 @@ func (c *Client) createKey(ctx context.Context, key Key) (*Key, error) {
 	return &keysResponse.Keys[0], nil
 }
 
+// TransferKeyToNewKeyRing method transfers a key associated with one key ring to another key ring
+// For more information please refer to the link below:
+// https://cloud.ibm.com/docs/key-protect?topic=key-protect-grouping-keys#transfer-key-key-ring
+func (c *Client) TransferKeyToNewKeyRing(ctx context.Context, keyID, newKeyRingID string) (*Key, error) {
```
Comment (mrodden) [inline keys.go:213]: This is more of a `SetKeyRing` honestly. It will always put the new value so its a set.

**Review by mrodden — CHANGES_REQUESTED** [review]

File: kp_test.go [hunk 3]
```diff
@@ -2700,6 +2700,116 @@ func TestGetKeyRings(t *testing.T) {
 	assert.True(t, gock.IsDone(), "Expected HTTP requests not called!")
 }
 
+func TestTransferKeyToNewKeyRing(t *testing.T) {
```
Comment (pbavari) [inline kp_test.go:2703]: Change this to match actual method name

File: kp_test.go [hunk 4]
```diff
@@ -2700,6 +2700,116 @@ func TestGetKeyRings(t *testing.T) {
...
+		Patch("/api/v2/keys/" + keyID).
+		Reply(200).
+		Body(bytes.NewReader((keyResponse)))
+
+	c, _, err := NewTestClient(t, nil)
+	gock.InterceptClient(&c.HttpClient)
+	defer gock.RestoreClient(&c.HttpClient)
+	c.tokenSource = &FakeTokenSource{}
+
+	key, err := c.TransferKeyToNewKeyRing(context.Background(), keyID, newKeyRingID)
```
Comment (pbavari) [inline kp_test.go:2747]: this should have not worked after changing method name.

**Review by pbavari — CHANGES_REQUESTED** [review]

File: keys.go [hunk 5]
```diff
@@ -203,6 +203,36 @@ func (c *Client) createKey(ctx context.Context, key Key) (*Key, error) {
...
+
+	if newKeyRingID == "" {
+		return nil, fmt.Errorf("Please provide a valid key ring id")
+	}
+
+	keyRingRequestBody := struct {
+		KeyRingID string
+	}{
+		KeyRingID: newKeyRingID,
+	}
```
Comment (mrodden) [inline keys.go:222]: Nice.

**Review by mrodden — APPROVED** [review]

**Review by pbavari — APPROVED** [review]

**Pull request merged**